src/share/import.js

```diff
@@ -0,0 +1,77 @@
+import fs from '../lib/fs';
+import Buffer from '../lib/buffer';
+import Path from '../lib/path';
+import WebTorrent from 'webtorrent';
+const sh = new fs.Shell();
+
+document.getElementById('torrentId').value = window.location.search.substring(
+    1
+);
+document.getElementById('btnDownload').addEventListener('click', startDownload);
+
+const downloadAllTorrentFiles = torrent =>
+    Promise.all(
+        torrent.files.map(
+            file =>
+                new Promise((resolve, reject) => {
+                    console.log('file: ', file);
+                    var filename = Path.join('/', file.path);
+                    file.getBuffer((err, buffer) => {
+                        if (err) {
+                            writeError();
+                            return reject(err);
+                        }
+                        sh.mkdirp(Path.dirname(filename), err => {
+                            if (err) {
+                                console.error(
+                                    'Error writing file, error was ',
+                                    err,
+                                    filename
+                                );
+                                writeError();
+                                return reject(err);
+                            }
+                            fs.writeFile(filename, new Buffer(buffer), err => {
+                                if (err) {
+                                    console.error(
+                                        'Error writing file, error was ',
+                                        err,
+                                        filename
+                                    );
+                                    writeError();
+                                    return reject(err);
+                                }
+                                console.log('Wrote to ', filename);
+                                resolve();
+                            });
+                        });
+                    });
+                })
+        )
+    );
+
+function startDownload() {

```

Comment:
Disable your button again after you click it, so we don't error on multiple clicks.  When the Torrent is complete, you can re-enable it so they can do another one if they like.